src/components/BibPage/BibDetail.tsx

```diff
@@ -0,0 +1,129 @@
+import {
+  Link as DSLink,
+  Heading,
+  List,
+} from "@nypl/design-system-react-components"
+import styles from "../../../styles/components/BibDetails.module.scss"
+import RCLink from "../RCLink/RCLink"
+import type {
+  BibDetail,
+  Url,
+  LinkedBibDetail,
+  SubjectHeadingDetail,
+} from "../../types/bibTypes"
+import { displayRtl, isItTheLastElement } from "../../utils/bibDetailUtils"
+import type { ReactNode } from "react"
+
+interface BibDetailsProps {
+  details: BibDetail[] | LinkedBibDetail[]
+  heading?: string
+}
+
+const BibDetails = ({ details, heading }: BibDetailsProps) => {
+  return (
+    details?.length > 0 && (
+      <List
+        title={heading && <Heading level="three">{heading}</Heading>}
+        noStyling
+        type="dl"
+        className={styles.bibDetails}
+        sx={{ borderBottom: "none" }}
+      >
+        {details.map(
+          (detail: BibDetail | LinkedBibDetail | SubjectHeadingDetail) => {
+            if (!detail) return
+            if (detail.label === "Subjects") {
+              return buildCompoundSubjectHeadingElement(
+                detail as SubjectHeadingDetail
+              )
+            } else if ("link" in detail) {
+              return buildLinkedElement(detail as LinkedBibDetail)
+            } else {
+              return buildPlainTextElement(detail as BibDetail)
+            }
+          }
+        )}
+      </List>
+    )
+  )
+}
+
+const buildDetailElement = (label: string, listChildren: ReactNode[]) => {
```

Comment:
Small nit, but what do you think of renaming these to DetailElement, PlainTextElement and CompoundSubjectHeadingElement? I know they're technically functions but I'd rather keep the naming consistent with other react elements.

Reply:
sounds good to me!